### PR TITLE
chore: bump Go MSRV to 1.22 and grocksdb version

### PR DESCRIPTION
**Pull request by faddat**
- **build(deps): bump github.com/linxGnu/grocksdb from 1.8.12 to 1.8.14**
- **update rocksdb in the dockerfile that the tests run in**

### CODE REVIEW

File: tools/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ RUN apt update \
        libleveldb-dev libleveldb1d
 
 FROM build AS install
-ARG ROCKSDB=8.9.1
+ARG ROCKSDB=8.11.3
```
Comment (adizere) [inline tools/Dockerfile:21]: looks good cf. https://github.com/linxGnu/grocksdb/releases/tag/v1.8.14

**Review by adizere — COMMENTED** [review]
thanks @faddat, looks good!

will let @melekes give the final review and merge this

**Comment by faddat** [comment]
The release that we use has to match the release in grocksdb, so we would want to use 13 for now

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**